api/swagger.yaml

```diff
@@ -2870,6 +2870,14 @@ definitions:
             description: "Run an init inside the container that forwards signals and reaps processes. This field is omitted if empty, and the default (as configured on the daemon) is used."
             type: "boolean"
             x-nullable: true
+          PidsLimit:
+            description: |
+              Tune a container's PIDs limit. Set `0` or `null` for unlimited.
+            type: "integer"
+            format: "int64"
+            x-nullable: true
```

Comment:
Does this need to be nullable?
For services we do not support partial updates (e.g. the client must send the entire service spec), this is different than the containers API.

Reply:
Hm, good point; but actually thinking of a scenario now that I'm not sure we're taking into account right now (also for other options)

- create service with API v1.x, using features of API v1.x
- use an older client (v1.x - 1) to update the service
- the older client doesn't know about features of v1.x, and will strip those fields
- the daemon detects the older API version, thus resets those fields


I'll have to check our logic, and make sure that on _update_, we restore the previous value of any field that was in the service spec, but that's not supported by the API version used to update.

Otherwise (for this particular feature), the PID limit will be removed when updating the service with an older client, which ain't nice